src/bin_editor/Oni2_editor.re

```diff
@@ -74,6 +76,70 @@ let cliOptions =
       },
     ~printVersion,
   );
+
+let initWorkingDirectory = () => {
+  let path =
+    switch (cliOptions.folder) {
+    | Some(folder) => folder
+    | None =>
+      switch (Store.Persistence.(Store.get(Global.workspace, Global.store))) {
+      | Some(path) => path
+      | None =>
+        switch (Revery.Environment.getWorkingDirectory()) {
+        | "/" => Sys.getenv_opt("HOME") |> Option.value(~default="/")
+        | path => path
+        }
+      }
+    };
+
+  Log.info("Startup: Changing folder to: " ++ path);
+  try(Sys.chdir(path)) {
+  | Sys_error(msg) => Log.error("Folder does not exist: " ++ msg)
+  };
+
+  path;
+};
+
+let createWindow = (~forceScaleFactor, ~workingDirectory, app) => {
+  let (x, y, width, height) =
+    Store.Persistence.(
+      {
+        let store = Workspace.storeFor(workingDirectory);
+
+        (
+          Store.get(Workspace.windowX, store),
+          Store.get(Workspace.windowY, store),
+          Store.get(Workspace.windowWidth, store),
+          Store.get(Workspace.windowHeight, store),
+        );
+      }
+    );
+
+  let window =
+    App.createWindow(
+      ~createOptions=
+        WindowCreateOptions.create(
+          ~forceScaleFactor,
+          ~maximized=false,
+          ~vsync=Vsync.Immediate,
+          ~icon=Some("logo.png"),
+          ~titlebarStyle=WindowStyles.Transparent,
+          // ~x,
+          // ~y,
+          ~width,
+          ~height,
+          (),
+        ),
+      app,
+      "Oni2",
+    );
+
+  Window.setPosition(window, x, y);
```

Comment:
Ah, so you're work on the SDL2 / Revery API will let us pass this directly in to avoid jank on window creation 👍

Reply:
Indeed, hoping so!